Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -208,6 +264,74 @@ void DiscusMultipleScatteringCorrection::getXMinMax(const Mantid::API::MatrixWor
   }
 }
 
+void DiscusMultipleScatteringCorrection::prepareStructureFactor() {
+  // avoid repeated conversion of bin edges to points inside loop by converting to point data
+  convertWsToPoints(m_SQWS);
+  // if S(Q,w) has been supplied ensure Q is along the x axis of each spectrum (so same as S(Q))
+  if (m_SQWS->getAxis(1)->unit()->unitID() == "MomentumTransfer") {
+    auto transposeAlgorithm = this->createChildAlgorithm("Transpose");
+    transposeAlgorithm->initialize();
+    transposeAlgorithm->setProperty("InputWorkspace", m_SQWS);
+    transposeAlgorithm->setProperty("OutputWorkspace", "_");
+    transposeAlgorithm->execute();
+    m_SQWS = transposeAlgorithm->getProperty("OutputWorkspace");
+  } else if (m_SQWS->getAxis(1)->isSpectra()) {
+    // for elastic set w=0 on the spectrum axis to align code with inelastic
+    auto newAxis = std::make_unique<NumericAxis>(std::vector<double>{0.});
+    newAxis->setUnit("DeltaE");
+    m_SQWS->replaceAxis(1, std::move(newAxis));
+  }
+  // avoid repeated conversion of bin edges to points inside loop by converting to point data
+  convertWsToPoints(m_SQWS);
+
+  // generate log of the structure factor to support gaussian interpolation
+  m_logSQ = m_SQWS->clone();
+
+  for (size_t i = 0; i < m_logSQ->getNumberHistograms(); i++) {
+    auto &ySQ = m_logSQ->mutableY(i);
+
+    std::transform(ySQ.begin(), ySQ.end(), ySQ.begin(), [](double d) -> double {
+      if (d == 0.)
+        return -20.0;
```

Comment:
What's the significance of -20 here, is it just an unphysical value?

Reply:
This bit of code is copied from the original Discus Fortran code. It's just a large negative value where exp(-20) is considered to be close to zero. It gets around generating an infinity when you take the log of a workspace containing zeroes. I think it deserves a comment or perhaps constant rather than just an arbitrary looking number in the code.